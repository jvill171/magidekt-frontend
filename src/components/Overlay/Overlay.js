
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faArrowLeft } from "@fortawesome/free-solid-svg-icons";
import "./Overlay.scss"


const Overlay = ({isOpen, onClose, children}) =>{
    return(
      <>
        {isOpen && (
          <div className="overlay">
            <div className="overlay__background" />
            <div className="overlay__container">
              <div className="overlay__controls__container">
                <div className="overlay__controls">
                  <button
                    className="overlay__close"
                    onClick={onClose}
                  > 
                    <FontAwesomeIcon icon={faArrowLeft} className="back_arrow"/>
                  </button>
                </div>
              </div>
              {children}
            </div>
          </div>
        )}
      </>
    )
}

export default Overlay;