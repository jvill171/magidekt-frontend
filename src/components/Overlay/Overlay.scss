.overlay__background {
    background-color: rgba(0, 0, 0, 0.5);
    width: 100vw;
    height: 100vh;
    position: fixed;
    top: 0;
    left: 0;
    z-index: 1000;
}

.overlay__container {
    background-color: white;
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    margin: auto;
    z-index: 1000;
    width: fit-content;
    height: fit-content;
    max-height: 90vh;
    max-width: 90vw;
    overflow: auto;
    border-radius: 5px;
}

.overlay__container > *{
    padding: 0 30px;
}

.overlay__controls {
    display: flex;
    justify-content: flex-start;
    align-items: center;
}

.overlay__controls__container {
    position: sticky;
    top: 0;
    left: 0;
    background: white;
    background: linear-gradient(white 80%, rgba(255,255,255,0.1));
    padding: 2px;
    z-index: 1000; 
}

// Back/Close button
.overlay__close {
    border: none;
    background-color: transparent;
    font-size: 30px;
    cursor: pointer;
    color: black;
    
    &:hover{ background-color: lightgray; }
}
