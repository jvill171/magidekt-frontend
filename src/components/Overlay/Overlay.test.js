import React from 'react';
import { render } from '@testing-library/react';
import Overlay from './Overlay';


// Smoke test 1
test('renders Overlay component without crashing', () => {
    const onCloseMock = jest.fn();
    render(
        <Overlay
            isOpen={true}
            onClose={onCloseMock}/>
    );
});
// Smoke test 2 - with child
test('renders Overlay component without crashing; With a child component', () => {
    const onCloseMock = jest.fn();
    render(
        <Overlay
            isOpen={true}
            onClose={onCloseMock}>
            {
                <>
                    <p>Test Child</p>
                </>
            }
        </Overlay>
    );
});

test('Overlay component matches snapshot', () => {
    const onCloseMock = jest.fn();
    const { asFragment } = render(
        <Overlay
            isOpen={true}
            onClose={onCloseMock} />
    );

    expect(asFragment()).toMatchSnapshot();
});
