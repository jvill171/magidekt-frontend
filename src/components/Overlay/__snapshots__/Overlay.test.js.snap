// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="overlay"
    id="overlay"
  >
    <div
      class="overlay__background"
    />
    <div
      class="overlay__container"
    >
      <div
        class="overlay__controls__container"
      >
        <div
          class="overlay__controls"
        >
          <button
            class="overlay__close"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-left back_arrow"
              data-icon="arrow-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
